core/src/main/java/org/apache/iceberg/actions/SizeBasedFileRewriter.java

```diff
@@ -258,6 +262,24 @@ protected long writeMaxFileSize() {
     return (long) (targetFileSize + ((maxFileSize - targetFileSize) * 0.5));
   }
 
+  protected PartitionSpec outputSpec() {
+    return table.specs().get(outputSpecId);
+  }
+
+  protected int outputSpecId() {
```

Comment:
I'm ok with this (and the method below having the same name mostly because the scopes are different)

Reply:
I'm ok to change the name for the private method to `extractOutputSpecId` or `populateOutputSpecId`. Kept it like that seeing other method names format in this class.

Reply:
yeah that is traditionally what we do, I think the naming here is fine but usually we don't have this kind of overlap.